homeassistant/components/trafikverket_weatherstation/sensor.py

```diff
@@ -197,29 +210,36 @@ class TrafikverketWeatherStation(SensorEntity):
 
     def __init__(
         self,
-        weather_api,
-        name,
-        sensor_station,
+        weather_api: TrafikverketWeather,
+        name: str,
+        sensor_station: str,
         description: TrafikverketSensorEntityDescription,
-    ):
+    ) -> None:
         """Initialize the sensor."""
         self.entity_description = description
         self._attr_name = f"{name} {description.name}"
+        self._attr_unique_id = f"trafikverket_{name}_{description.key}"
```

Comment:
This is not unique enough, please include the config entry ID in the unique ID

Reply:
It contains the key which is unique for each sensor and same name can not be used twice so why would this not be unique?
I can of course add the entry id but it's not going to be any more unique than it already is.

Reply:
Why can't the same name be used twice, do we somehow block that in the config flow?

Reply:
Sorry, was mixing with imports. You are absolutely correct so I will update it.

Reply:
For clarity I had a block on using the same station twice but someone said "we are not to judge if someone wants to have same station twice" so hence that block in the config flow has since been taken off.
The block now only remains in import to stop importing each time.

Reply:
Right, so the `sensor_station` is not guaranteed to be unique.
Hence, pass the config entry to the constructor, and do:
```py
self._attr_unique_id = f"{config_entry.entry_id}_{description.key}"
```